### PR TITLE
Encode pickled primitives directly using reflection

### PR DESCRIPTION
**Pull request by nicolasstucki**
This removes the need of some extra helper methods in the internal package.

Add `asExpr` to have a variant of `asExprOf[Any]` that does not need the extra type test at runtime.
Also add an equivalent `asType` method.

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Step towards #10222

File: library/src/scala/tasty/Reflection.scala
Comment (smarter) [inline library/src/scala/tasty/Reflection.scala:1786]: what's the difference with `seal`?

Comment (nicolasstucki) [reply library/src/scala/tasty/Reflection.scala:1786]: It will replace `seal`. But that needs to be removed after #10207 is merged. In that PR we remove the `unseal` method from type. I decided to do it this way to avoid blocking this PR.

Comment (liufengyun) [reply library/src/scala/tasty/Reflection.scala:1786]: I find the name `asType` is not as informative as `seal`, because the name `Type` in the context of `TypeRepr` is vague, it could refer to both `TypeRepr` and `quoted.Type`.

**Review by liufengyun — APPROVED** [review]
LGTM

File: library/src/scala/tasty/Reflection.scala
Comment (nicolasstucki) [reply library/src/scala/tasty/Reflection.scala:1786]: The next step is to find a way to inject this type in the context directly. Something like

```scala
typeRep.asTypeInContext { [T] =>>
   // Type[T] is given in this context
}

def asTypeInContext[U](body: [T] =>> (using Type[T]) => U): U = ...
```

**Pull request merged**
